jwst/ramp_fitting/ramp_fit.py

```diff
@@ -247,53 +254,71 @@ def ols_ramp_fit(model, buffsize, save_opt, readnoise_model, gain_model,
 
             data_sect = model.get_section('data')[num_int, :, rlo:rhi, :]
 
-            # skip data section if it is all NaNs
+            # Skip data section if it is all NaNs
             if  np.all(np.isnan( data_sect)):
                 log.error('Current data section is all nans, so not processing the section.')
-                break   
-
-            # Poisson noise only; read noise only
-            var_p_2d = np.zeros((max_seg, (rhi - rlo) * imshape[-1]),
-                                dtype=np.float32)
-            var_r_2d = np.zeros((max_seg, (rhi - rlo) * imshape[-1]),
-                                dtype=np.float32)
+                break
```

Comment:
It looks to me as if this would exit the loop over data sections, i.e. there would be no further processing of the data.  Isn't `continue` what you want, rather than `break`?

Reply:
Yes, switched to continue.